foodanddrink/restaurant/views.py

```diff
@@ -81,3 +83,15 @@ def profile(request):
     'profile': customer
   }
   return render(request, 'restaurant/profile.html', context)
+# def order(request):
+#   return render(request,'restaurant/checkout.html')
+
+def order_detail_view(request, pk):
+  customer_order = Order.objects.get(pk=pk)
+  items_in_cart = OrderDetail.objects.filter(order=pk).select_related("product")
+  return render(request,'restaurant/checkout.html', context={'customer_order': customer_order, 'items_in_cart': items_in_cart})
+
+def delete_a_product(request, pk, pk2):
+    deleted_product = OrderDetail.objects.filter(order=pk).filter(pk=pk2)
```

Comment:
check trường hợp ko tìm thấy nữa em

Reply:
Ở trong bảng hiển thị list ds các sp người dùng đã order (cart):
-Chức năng xóa sp: rõ ràng sp phải có tồn tại thì mới hiển thị để xóa được nên em nghĩ không cần ạ

Reply:
nhỡ có giao dịch xóa sản phẩm xảy ra đồng thời và trước khi thanh toán nữa e nhé nên luôn xử lý để an toàn cho ứng dụng

Reply:
dạ vâng em đang hoàn thiện ở pull tiếp theo đây ạ! Cái này em lỡ code để test thôi ạ :)